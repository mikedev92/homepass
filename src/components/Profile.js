// @flow
import React from 'react';
import { connect } from 'react-redux';
import { Header, Left, Container, Content, Icon, Button, Text } from 'native-base';
import { withNavigation } from 'react-navigation';
import { compose } from 'recompose';
import * as fromActions from '../actions/auth';

type Props = {
  logout: () => void,
  navigation: Navigation,
};

function Profile({ logout, navigation }: Props) {
  return (
    <Container>
      <Header>
        <Left>
          <Button transparent primary onPress={() => navigation.goBack()}>
            <Icon name="arrow-back" />
          </Button>
        </Left>
      </Header>
      <Content style={{ backgroundColor: 'white' }}>
        <Text>Profile</Text>
        <Button onPress={logout}><Text>Logout</Text></Button>
        <Button onPress={() => navigation.navigate('StyleGuide')}><Text>Style Guide</Text></Button>
      </Content>
    </Container>
  );
}

Profile.navigationOptions = {
  title: 'Profile',
};

export default compose(
  withNavigation,
  connect(null, (dispatch: Dispatch) => ({
    logout: () => dispatch(fromActions.logout()),
  })),
)(Profile);
