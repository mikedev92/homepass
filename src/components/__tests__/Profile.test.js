// @flow
import React from 'react';
import { shallow } from 'enzyme';
import Profile from '../Profile';

describe('<Profile />', () => {
  it('matches snapshots', () => {
    const wrapper = shallow(<Profile />);
    expect(wrapper).toMatchSnapshot();
  });
});
