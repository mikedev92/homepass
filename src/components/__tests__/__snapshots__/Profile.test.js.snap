// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshots 1`] = `
<Styled(Container)>
  <Styled(Header)>
    <Styled(Left)>
      <Styled(Button)
        onPress={[Function]}
        primary={true}
        transparent={true}
      >
        <Styled(Icon)
          name="arrow-back"
        />
      </Styled(Button)>
    </Styled(Left)>
  </Styled(Header)>
  <Styled(Content)
    style={
      Object {
        "backgroundColor": "white",
      }
    }
  >
    <Styled(Text)>
      Profile
    </Styled(Text)>
    <Styled(Button)>
      <Styled(Text)>
        Logout
      </Styled(Text)>
    </Styled(Button)>
    <Styled(Button)
      onPress={[Function]}
    >
      <Styled(Text)>
        Style Guide
      </Styled(Text)>
    </Styled(Button)>
  </Styled(Content)>
</Styled(Container)>
`;
